### PR TITLE
Base: Improve default WebServer pages

### PR DESCRIPTION
**Pull request by electrikmilk**
Add some fancy stuff like semantic tags, styling and the lady ball to the default web server pages :^).

Inspired by Apache Ubuntu/Debian default index.html.

Before:
Screen Shot 2022-02-21 at 21 36 18

After:
Screen Shot 2022-02-21 at 21 44 20

### CODE REVIEW

**Comment by drunderscore** [comment]
(on mobile, can't properly review) The "last modified" HTML comments seem a bit superfluous, we have git commit history for this :^)

**Comment by paulwratt** [comment]
does Serenity have a "default color palette" ? Does Browser have a @native GUI CSS tag yet?

**Comment by paulwratt** [comment]
Whats with all the `<header>`, `<main>` and `<section>` tags.  With only `<!doctype html>` and `<html lang="en">` those tags are invalid.

Now I cant browser the Serenity Web Server with my HTML 3.2 Browser ..

Also you forgot the `</head>` tag.

**Comment by linusg** [comment]
> Whats with all the <header>, <main> and <section> tags. With only <!doctype html> and <html lang="en"> those tags are invalid.

No idea what you're talking about, `<!doctype html>` is the HTML5 doctype.

> Now I cant browser the Serenity Web Server with my HTML 3.2 Browser ..

https://github.com/SerenityOS/serenity/pull/5191#issuecomment-773873994

**Comment by paulwratt** [comment]
there a difference between _inclusion_ and _exclusion_ . Are you saying WebServer _is not_ going to support HTTP/1.0 ?

**Comment by paulwratt** [comment]
> > Whats with all the `<header>`, `<main>` and `<section>` tags.  With only `<!doctype html>` and `<html lang="en">` those tags are invalid.
> 
> No idea what you're talking about, `<!doctype html>` is the HTML5 doctype, and those are perfectly valid.
> 

And _still_ inapropriate for the "Well Hello Friends" _landing page_ for your "now working, SerenityOS WebServer". (`HTML5 !== HTML`).

I could have said ".. those tags are invalid for non-HTML5 browsers, without the addition of a 1st line _schema_  or ENTITY tag (comparible to the HTML 4+ _transitional_ tags)." ([Browser _section_ tag support is limited](https://www.w3schools.com/TAgs/tag_section.asp#:~:text=in%20a%20document.-,Browser%20Support,-The%20numbers%20in)) - but I thought everyone would know that was what I meant.

> > Now I cant browser the Serenity Web Server with my HTML 3.2 Browser ..
> 
> [#5191 (comment)](https://github.com/SerenityOS/serenity/pull/5191#issuecomment-773873994)

Then _why_ include [those 2 meta tags](https://stackoverflow.com/a/51006700/8268081).

You do realise the _context_ in which [boricj made that comment](https://github.com/SerenityOS/serenity/pull/5191#issue-797550086), right?

And even without that context, Serenity(OS) has to _support archiology_, by virtue of what it is _and_ what it is doing, even if it itself is _not about archeology_.

---

Anyway, my point was to rather to cause _think _latterally_. Those tags dont _have_ to be used (why not `div` and `span`, or `center`, `i` and `b` - with ID/Class, which can _also_ be rendered by non-CSS Browsers), they are not essential to its functionality, which is why the originals were _super generic_, the same goes for the CSS, (there are not, but I am just saying) dont use the _fancy ones_, while showing continuity with Browser, HTML & CSS _Fu_ (lets not _try_ to break the friendship of our "newly aquired friends, however old they are").

@electrikmilk if you wrap your new "box" layout with the following, it _will_ center any internal content in its _compacted form_ without forcing the its inner text to be _center aligned_, and show in the very _center_ of the browser window.

It works 100% in things like IE3, NS2, and the many HTML 3.2 & non-JS Web Browsers still being used out there today.
```
<table border=0 cellpadding=0 cellspacing=0 align=center valign=middle width=100% height=100%><tr align=center valign=middle width=100% height=100%><td align=center valign=middle width=100% height=100%><table><tr><td>
 ...
</td></tr></table></td></tr></table>
``

**Comment by linusg** [comment]
This is not up for discussion, we're not going to use table layouts or make the HTML worse to deal with ancient browsers. 

@electrikmilk New page looks very nice, please fix up that missing head tag and I think it's good to go.

**Comment by paulwratt** [comment]
@linusg I guess I'll have to remember that next time I start up SerentyOS - sorry.

**Comment by electrikmilk** [comment]
> (on mobile, can't properly review) The "last modified" HTML comments seem a bit superfluous, we have git commit history for this :^)

That's perfectly fair, I was going off of the Apache default page which has this 🤦

**Comment by electrikmilk** [comment]
> Whats with all the `<header>`, `<main>` and `<section>` tags. With only `<!doctype html>` and `<html lang="en">` those tags are invalid.
> 
> Now I cant browser the Serenity Web Server with my HTML 3.2 Browser ..
> 
> (I was going to share a trick to get that box to render centered on _any_ browser, incl. 3.2 non-JS ones)
> 
> Also you forgot the `</head>` tag.

@paulwratt @linusg I can't believe I forgot the `<head>` tag, I'll fix that.

**Comment by electrikmilk** [comment]
> 

That's a fair point, I will fix and use `<meta charset="utf-8">` instead as it's the shortest most modern way.

Serenity is more about the *look* from what I understand, but with modern guts, so I'm not looking to use the `<center>` tag or tables because the `<center>` tag is (deprecated)[https://developer.mozilla.org/en-US/docs/Web/HTML/Element/center] and tables are not a modern way of organizing content, we are looking to make our technology and usage as modern as possible. I think it's laudable that the page shouldn't be too complicated.

**Comment by electrikmilk** [comment]
Removed `<!-- Last Modified... -->` comment and ended `<head>` tag properly in latest commit.

File: Base/www/index.html [hunk 1]
```diff
@@ -1,18 +1,66 @@
...
+        in your browser (cheater!) :^)
+      </p>
+      <p>
+        Try this <a href="other.html">link</a> to another page!
+      </p>
+    </section>
+  </main>
+</body>
+
+</html>
```
Comment (linusg) [inline Base/www/index.html:66]: Please configure your editor to add a newline to the end of files

File: Base/www/other.html [hunk 2]
```diff
@@ -1,16 +1,63 @@
...
+        This is not even index.html, how neat is that! :^)
+      </p>
+      <p>
+        Maybe you want to go <a href="index.html">back to index.html</a>?
+      </p>
+    </section>
+  </main>
+</body>
+
+</html>
```
Comment (linusg) [inline Base/www/other.html:63]: Please configure your editor to add a newline to the end of files

File: Base/www/other.html [hunk 3]
```diff
@@ -1,16 +1,63 @@
...
+    }
+
+    .banner.it-works {
+      background: #ab6e4aff;
+    }
+
+    section {
+      margin: 5px 0;
+    }
+  </style>
```
Comment (linusg) [inline Base/www/other.html:41]: head tag still missing here

**Review by linusg — CHANGES_REQUESTED** [review]

**Comment by electrikmilk** [comment]
Should be all fixed in latest push.

**Review by linusg — CHANGES_REQUESTED** [review]
Now the indentation changed from spaces to tabs for some reason :neutral_face:
Please just run prettier over them?

**Comment by electrikmilk** [comment]
> Now the indentation changed from spaces to tabs for some reason :neutral_face:
> 
> Please just run prettier over them?

I had reformatted, I didn't know spaces were preferred, sorry!

**Comment by linusg** [comment]
I would've complained if spaces wasn't right :^)

**Review by linusg — APPROVED** [review]
Awesome, thanks!

**Comment by electrikmilk** [comment]
> I would've complained if spaces wasn't right :^)

That's fair, used a different text editor to try and add a new line to the end of them, I'll remember not to reformat using that editor in the future lol.

**Pull request merged**